### PR TITLE
Hammy dialogue

### PR DESCRIPTION
**Pull request by winka826**
User input error handling implemented
Almost all daily log functions
Saves user input to journal class 
Menu prints - but does not have functionality (other than daily log)

### CODE REVIEW

File: Hustle-Hamster/main.cpp [hunk 1]
```diff
@@ -19,12 +22,18 @@
...
 
-int dayRating = 0; 
-char newLine[] = "\n\n";
+/* The standard delay used for printing */
 unsigned int stdDelay = 10;
-vector<string> completedActivities; 
+/* The Inital default activities */
+vector<string> defaultActivities = {"Study", "Work", "Socialise", "Exercise", "Drink Water", "Go outside"}; 
+/* Separator for formatting */
 char separater[] = "----------------------------------------------------------";
```
Comment (mary0L) [inline Hustle-Hamster/main.cpp]: I think this might be spelt separator?

File: Hustle-Hamster/main.cpp [hunk 2]
```diff
@@ -36,86 +44,211 @@ int printHammy(){
...
+    while(!validResponse){
+        cin >> temp;
+        if(!cin.fail() && (temp>0 && temp<6)){
+            dailyEntry.setDayRating(temp);
+            validResponse = true;
+            break;
+        }else{
+            TYPE("Please enter a valid number between 1 and 5");
+            cin.clear();
+            cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
```
Comment (arcfernandes04) [inline Hustle-Hamster/main.cpp]: `max()` function used here causes:
`Warning C4003 not enough arguments for function-like macro invocation 'max'`

**Review by arcfernandes04 — CHANGES_REQUESTED** [review]
I've left a comment on the specific line causing issues, but could you also pull the updates from main into this branch before PR :)

File: Hustle-Hamster/main.cpp (hunk 2 above)
Comment (winka826) [reply Hustle-Hamster/main.cpp]: OK I think I have fixed it here, it just means that for the string there is a certain buffer size, and just need to be careful that user input won't exceed the buffer. Or find an alternative later down the line.

**Pull request merged**